Document analysis strategies by dynamic vs static (#1036)
fossas/fossa-cli: docs/references/strategies/README.md

- [carthage](platforms/ios/carthage.md)
- [cocoapods](platforms/ios/cocoapods.md)
- [swiftPM](platforms/ios/swift.md)

## Strategies by type

> Languages supported by FOSSA CLI can have multiple strategies for detecting dependencies, one primary strategy that yields ideal results and zero or more fallback strategies. Within this list of strategies, we have the concept of *static* and *dynamic* strategies. Static strategies parse files to find a dependency graph (example: parse a `package-lock.json` file). Dynamic strategies are required when analyzing package managers that do not offer complete lockfiles, such as Gradle or Golang. Dynamic strategies require a working build environment to operate in.

Reviewer: [nit] The name of the language is [Go, not Golang](https://go.dev/doc/faq#go_or_golang).
```suggestion
Languages supported by FOSSA CLI can have multiple strategies for detecting dependencies, one primary strategy that yields ideal results and zero or more fallback strategies. Within this list of strategies, we have the concept of *static* and *dynamic* strategies. Static strategies parse files to find a dependency graph (example: parse a `package-lock.json` file). Dynamic strategies are required when analyzing package managers that do not offer complete lockfiles, such as Gradle or Go. Dynamic strategies require a working build environment to operate in.
```

Author: There are probably a lot more errors out there like this. I always type "Golang" because "Go" is honestly sometimes confusing.

Author: I think the court of public opinion is split https://hub.docker.com/r/circleci/golang, but I'll change the references I can find.

> > If the FOSSA CLI uses a fallback strategy and does not detect ideal results, you will see a "Warning" in the output when running `fossa analyze`.

Reviewer: [nit] In technical writing, prefer to describe facts instead of speaking to the user directly. 

Similarly, prefer to be explicit when a causal chain exists. "It used a fallback strategy, therefore its results are not ideal" is more explicit about the causal relationship than "it used a fallback strategy and its results were not ideal"- the latter leaves room for the interpretation that it is possible to have ideal results with a fallback strategy, and then implies that in such a case no warning will be emitted.
```suggestion
> If the FOSSA CLI is forced to utilize a fallback strategy, meaning it did not detect ideal results, a warning is emitted in the scan summary after running `fossa analyze`.
```

> \* - denotes the primary strategy (all other strategies are fallbacks)

Reviewer: [optional] I recommend having a third column, "Primary", which then contains one of the literals "Dynamic" or "Static". I think this makes it more clear.

| ----------------------------------------------------------------------------------------------------------------------------------------------- | ------- | ------ |
| [C#](https://github.com/fossas/fossa-cli/tree/master/docs/references/strategies/languages/dotnet)                                               | ✅\*    | ✅     |
| [Clojure (leiningen)](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/clojure/clojure.md)                  | ✅\*    | ❌     |
| [Dart (pub)](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/dart/dart.md)                                 | ✅\*    | ✅     |
| [Elixer (mix)](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/elixir/elixir.md)                           | ✅\*    | ❌     |
| [Erlang (rebar3)](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/erlang/erlang.md)                        | ✅\*    | ❌     |
| [Fortran](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/fortran/fortran.md)                              | ❌      | ✅\*   |

> | [Golang (dep)](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/golang/godep.md)                            | ❌      | ✅\*   |
> | [Golang (glide)](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/golang/glide.md)                          | ❌      | ✅\*   |
> | [Golang (gomodules)](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/golang/gomodules.md)                  | ✅\*    | ✅     |

Reviewer: ```suggestion
| [Go (dep)](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/golang/godep.md)                            | ❌      | ✅\*   |
| [Go (glide)](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/golang/glide.md)                          | ❌      | ✅\*   |
| [Go (gomodules)](https://github.com/fossas/fossa-cli/blob/master/docs/references/strategies/languages/golang/gomodules.md)                  | ✅\*    | ✅     |
```

- [carthage](platforms/ios/carthage.md)
- [cocoapods](platforms/ios/cocoapods.md)
- [swiftPM](platforms/ios/swift.md)

## Strategies by type

Languages supported by FOSSA CLI can have multiple strategies for detecting dependencies, one primary strategy that yields ideal results and zero or more fallback strategies. Within this list of strategies, we have the concept of _static_ and _dynamic_ strategies. Static strategies parse files to find a dependency graph (example: parse a `package-lock.json` file). Dynamic strategies are required when analyzing package managers that do not offer complete lockfiles, such as Gradle or Golang. Dynamic strategies require a working build environment to operate in.

> It is important to note that neither type of strategy has an inherent benefit when detecting dependencies. If a supported language does not have static and dynamic strategies, this does not mean it is less supported than a language that does.

Reviewer 2: ```suggestion
It is important to note that neither type of strategy has an inherent benefit when detecting dependencies. If a supported language has only a static or only a dynamic strategy, this does not mean it is less supported than a language that
```

Reviewer 2: This suggestion reads a bit more clearly to me because I could read the earlier version as saying a language that has neither a static nor a dynamic strategy might be well supported.
